src/main/java/com/pinterest/secor/common/SecorConfig.java

```diff
@@ -211,6 +211,10 @@ public String getMessageTimestampInputPattern() {
         return getString("message.timestamp.input.pattern");
     }
 
+    public int getFinalizerLookbackPeriods() {
+        return getInt("secor.finalizer.lookback.periods", 10);
```

Comment:
You should put this param in one of the .properties files and add some documentation. You can comment it out by default.